highlighters/main/main-highlighter.zsh

```diff
@@ -406,6 +406,8 @@ _zsh_highlight_main_highlighter_highlight_list()
   # - :regular:    "Not a command word", and command delimiters are permitted.
   #                Mainly used to detect premature termination of commands.
   # - :always:     The word 'always' in the «{ foo } always { bar }» syntax.
+  # - :test:       Between [[ and ]]
+  # - :end:        Must be a command separator; anything else is a syntax error.
```

Comment:
States say what's allowed; there's no precedent of a state saying what's _disallowed._ Thus, the part after the semicolon is misleading, I think: it's only accurate in _this_ use case, but not in  `:foo::end:` (which may happen in the future).